hal-core/src/addr.rs

```diff
@@ -100,7 +100,7 @@ pub trait Address:
 
     /// Returns `true` if `self` is aligned on the specified alignment.
     fn is_aligned<A: Into<usize>>(self, align: A) -> bool {
-        self.align_down(align) == self
+        self.as_usize() % align.into() == 0
```

Comment:
yeah, that's probably a good call. we probably should have a debug assertion that the alignment is a power of two, though, since this accepts an arbitrary number as an alignment...

Reply:
i think in the version that checks against the align of a type we can probably rely on rustc not making types with weird alignment